artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/impl/BridgeImpl.java

```diff
@@ -467,7 +433,7 @@ public void sendFailed(Message message, Exception e) {
    public void sendAcknowledged(final Message message) {
       logger.trace("BridgeImpl::sendAcknowledged received confirmation for message {}", message);
 
-      if (active) {
+      if (state == State.STARTED || state == State.STOPPING || state == State.PAUSING) {
```

Comment:
Given the variable is now volatile, might be nicer to read the value once to a local variable and operate on that in this likely active section?

I also wonder if, given the logging above this line, whether there should be some more logging to note when it entirely skips doing anything here.

Reply:
Agreed on both points.